modules/aws/lambda.go

```diff
@@ -2,51 +2,136 @@ package aws
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 
 	"github.com/aws/aws-sdk-go/service/lambda"
 	"github.com/gruntwork-io/terratest/modules/testing"
 	"github.com/stretchr/testify/require"
 )
 
+// LambdaOptions contains additional parameters for InvokeFunctionWithParams().
+// It contains a subset of the fields found in the lambda.InvokeInput struct.
+type LambdaOptions struct {
+	// FunctionName is a required field containing the lambda function name.
+	FunctionName *string
+
+	// InvocationType can be one of "RequestResponse" or "DryRun".
+	//    * RequestResponse (default) - Invoke the function synchronously.
+	//    Keep the connection open until the function returns a response
+	//    or times out.
+	//
+	//    * DryRun - Validate parameter values and verify that the user or
+	//    role has permission to invoke the function.
+	InvocationType *string
```

Comment:
NIT: Any reason not to use the AWS enum type here? Reduces error checking down below.

Reply:
To be frank, I'm having a hard time figuring out what part of the AWS SDK you're trying to "shield" from the user and what part you're okay with exposing.  So I try something I think is inline with your approach and then adjust when you tell me I'm wrong.  It's not the best use of time for either of us.

Done

Reply:
That still seems cleaner than having an untyped string though, right?